### PR TITLE
#220 | Refactor Topic & Subscription State Management and Improve API Idempotency

### PR DESCRIPTION
**Pull request by devanshu-garg-IN**
# Description [#220]

## Key Changes

# Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

# Testing and Validation

- Added comprehensive unit tests to verify the correctness of the implementation.
- Included edge cases to ensure robust behavior.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: entities/src/main/java/com/flipkart/varadhi/entities/AbstractTopic.java [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 
+/**
+ * Abstract class representing a topic entity in the MetaStore.
+ * Extends the MetaStoreEntity class.
+ */
 public abstract class AbstractTopic extends MetaStoreEntity {
 
-    public AbstractTopic(String name, int version) {
+    /**
+     * Constructor to initialize the AbstractTopic with a name and version.
```
Comment (gauravAshok) [inline entities/src/main/java/com/flipkart/varadhi/entities/AbstractTopic.java:10]: any reason we are adding docs for little things ?

Such docs dont help with much. It doesn't contain "context".

**Comment by gauravAshok** [comment]
changes looks fine largely.

File: entities/src/main/java/com/flipkart/varadhi/entities/AbstractTopic.java (hunk 1 above)
Comment (devanshu-garg-IN) [reply entities/src/main/java/com/flipkart/varadhi/entities/AbstractTopic.java]: Understood! The intent behind adding Javadocs was to maintain consistency and provide at least a basic level of documentation. However, I see the point about lacking context. Moving forward, I’ll ensure that Javadocs include relevant context else will try to avoid it. Let me know if there are specific guidelines you’d like to follow to make them more useful.

Comment (gauravAshok) [reply entities/src/main/java/com/flipkart/varadhi/entities/AbstractTopic.java]: I will try to come up with guidelines.
